polaris-core/src/main/java/org/apache/polaris/core/config/FeatureConfiguration.java

```diff
@@ -83,6 +83,7 @@ public static void enforceFeatureEnabledOrThrow(
           .defaultValue(false)
           .buildFeatureConfiguration();
 
+  @SuppressWarnings("deprecation")
```

Comment:
This does get rid of the warnings, but aren't the warnings the main feature of the @Deprecated annotation?

ref: https://github.com/apache/polaris/pull/1672#discussion_r2107737093

Reply:
Thanks for the link I didn't know there was a previous conversation on this topic.

That said, I rather disagree: the annotation is valuable mainly for consumers of the API, and especially when it's well documented in terms of when it was deprecated, when it will be removed and how to replace it. 

Here, we own both the source of the deprecation warnings (the `catalogConfigUnsafe` method) and the call sites where the deprecation warnings are triggered (in `FeatureConfiguration` class): IOW, keeping the warnings around is not useful for Polaris devs.

And there is no risk of "forgetting" to update `FeatureConfiguration` later on: as soon as we remove the deprecated method, the `FeatureConfiguration` class wouldn't compile anymore.

Reply:
My take on this: we should rather remove deprecation from `catalogConfigUnsafe()` and file issues to remove support for those legacy properties. The method itself has a valid use case in Polaris code, which is specifically to support migration from old config names to new name, so the method itself is not conceptually "deprecated".

Reply:
A `io.smallrye.config.RelocateConfigSourceInterceptor` implementation can be used to nag users to update the configs.

Reply:
There's already a log to remind users to remove the configs, so we are covered from that perspective. This was more intended to nag maintainers to remove the deprecated method at some point in time. Otherwise, why deprecate methods?